driftbase/match_placements.py

```diff
@@ -0,0 +1,539 @@
+
+            lobby_lock.lobby = lobby
+
+            log.info(f"GameLift game session placement issued for lobby '{lobby_id}' by host player '{player_id}'")
+
+            # Notify members
+            receiving_player_ids = _get_lobby_member_player_ids(lobby, [player_id])
+            _post_lobby_event_to_members(receiving_player_ids, "LobbyMatchStarting", {"lobby_id": lobby_id, "status": lobby["status"]})
+
+            return match_placement
+
+def stop_player_match_placement(player_id: int, expected_match_placement_id: str):
+    with _GenericLock(_get_player_match_placement_key(player_id)) as player_match_placement_lock:
+        placement_id = player_match_placement_lock.value
+
+        if expected_match_placement_id != placement_id:
+            log.warning(f"Player '{player_id}' attempted to stop match placement '{expected_match_placement_id}', but the player didn't issue the match placement")
+            raise UnauthorizedException(f"You don't have permission to access match placement {expected_match_placement_id}")
+
+        with _JsonLock(_get_match_placement_key(placement_id)) as match_placement_lock:
+            placement = match_placement_lock.value
+
+            if placement:
+                lobby_id = placement.get("lobby_id", None)
+                if not lobby_id:
+                    raise RuntimeError(f"Match placement '{placement_id}' doesn't contain a 'lobby_id' field. Only lobby match placements are supported at this time")
+
+                match_provider = placement["match_provider"]
+                if match_provider != "gamelift":
+                    raise RuntimeError(f"Invalid match provider configured, '{match_provider}'. Only the GameLift match provider is supported at this time")
+
+                placement_status = placement["status"]
+                if placement_status != "pending":
+                    log.warning(f"Player '{player_id}' attempted to stop match placement '{expected_match_placement_id}', but the placement is in status '{placement_status}'")
+                    raise InvalidRequestException(f"Cannot stop a match placement in status {placement_status}")
+
+                response = flexmatch.stop_game_session_placement(placement_id)
+
+                log.debug(f"match_placements::stop_player_match_placement() stop_game_session_placement response: '{_jsonify(response)}'")
+
+                log.info(f"Player '{player_id}' stopped match placement '{placement_id}'")
+
+                match_placement_lock.value = None
+
+        player_match_placement_lock.value = None
+
+def process_gamelift_queue_event(queue_name: str, message: dict):
+    log.debug(f"match-placements::process_gamelift_queue_event() received event in queue '{queue_name}': '{message}'")
+
+    event_details = _get_event_details(message)
+    event_type = event_details.get("type", None)
+    if event_type is None:
+        raise RuntimeError(f"No event type found. Message: '{message}'")
+
```

Comment:
Shouldn't we somehow filter out queue events not related to lobbies here?

Reply:
It is filtered out later in `_validate_gamelift_placement_for_queue_event`

If it can't find a match placement for the placement id, then it ignores it.